src/Agent.Plugins/PipelineArtifact/PipelineArtifactPlugin.cs

```diff
@@ -36,6 +42,13 @@ public async Task RunAsync(AgentTaskPluginExecutionContext context, Cancellation
             await ProcessCommandInternalAsync(context, targetPath, artifactName, token);
         }
 
+        private string NormalizeJobIdentifier(string jobIdentifier)
+        {
+            Regex rgx = new Regex("[^a-zA-Z0-9 - .]");
+            jobIdentifier = rgx.Replace(jobIdentifier, string.Empty).Replace(".default", string.Empty);
+            return jobIdentifier;
```

Comment:
Please label in a comment as to what this is doing, it's not entirely clear as to what this was doing. As for the regex, this is appropriate for this logic.